man/tpm2_getekcertificate.1.md

```diff
@@ -0,0 +1,63 @@
+% tpm2_getekcertificate(1) tpm2-tools | General Commands Manual
+
+# NAME
+
+**tpm2_getekcertificate**(1) - Retrieve the Endorsement key Certificate for the TPM
+endorsement key from the TPM manufacturer's endorsement certificate hosting
+server.
+
+# SYNOPSIS
+
+**tpm2_getekcertificate** [*OPTIONS*] _EK\_PUBLIC\_FILE_
+
+# DESCRIPTION
+
+**tpm2_getekcertificate**(1) - Retrieve the Endorsement key Certificate for
+the TPM endorsement key from the TPM manufacturer's endorsement certificate hosting
+server.
+
+# OPTIONS
+
+  * **-o**, **\--ek-certificate**=_EK\_CERTIFICATE\_FILE_:
+
+    Specifies the file used to save the Endorsement key certificate retrieved from
+    the TPM manufacturer provisioning server. Defaults to stdout if not
+    specified.
+
+  * **-X**, **\--allow-unverified**:
+
+    Specifies to attempt connecting with the TPM manufacturer provisioning server
+    without verifying server certificate.
+
+    **WARNING**: This option should be used only on platforms with older CA certificates.
+
+  * **-a**, **\--ekcertificate-portal**: _URL_
+
+    Specifies the url for the ek certificate portal.
+
+[common options](common/options.md)
+
+[common tcti options](common/tcti.md)
+
+[authorization formatting](common/authorizations.md)
+
+[supported public object algorithms](common/object-alg.md)
+
+[algorithm specifiers](common/alg.md)
+
+# NOTES
+
+When the verbose option is specified, additional curl debugging information is
+provided by setting the curl mode verbose, see
+<https://curl.haxx.se/libcurl/c/CURLOPT_VERBOSE.html> for more information.
+
+# EXAMPLES
+
+```
+tpm2_getekcertificate -X -o ECcert.bin ek.bin -a https://tpm.manufacturer.com/ekcertserver/
```

Comment:
full working example

Reply:
This will be an interesting one. In that, this would never work on a simulator. Nevertheless, I get your point. The example at least need to show how you arrive at the EK public required as an input to be given to the tool.

Reply:
Updated example section